### PR TITLE
feat: shipping details

### PR DESCRIPTION
**Pull request by kibo-sushant**

### CODE REVIEW

File: components/common/AddressForm/AddressForm.tsx [hunk 1]
```diff
@@ -87,10 +86,11 @@ const AddressForm = (props: AddressFormProps) => {
     setAutoFocus = true,
     validateForm = false,
     onSaveAddress,
-    onCompleteCallback,
     setValidateForm,
   } = props
 
+  const { setStepStatusIncomplete } = useCheckoutStepContext()
```
Comment (kibo-kevinwatts) [inline components/common/AddressForm/AddressForm.tsx:92]: Won't this put a hard dependency on Checkout for all address forms? How will we reuse for "My Account" forms (or anywhere else)

File: hooks/queries/useShippingMethods/useShippingMethods.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  return response?.orderShipmentMethods
+}
+
+export const useShippingMethods = (checkoutId: string): UseShippingMethodsResponse => {
+  const {
+    data = [],
+    isLoading,
+    isSuccess,
+  } = useQuery(shippingMethodKeys.detail(checkoutId), () => loadShippingMethods(checkoutId))
```
Comment (kibo-kevinwatts) [inline hooks/queries/useShippingMethods/useShippingMethods.tsx:31]: We should  disable caching for this request. There are factors outside of our parameters (shipping provider) that can affect the available shipping methods.

File: lib/react-query/queryKeys.ts [hunk 3]
```diff
@@ -5,6 +5,11 @@ export const checkoutKeys = {
   detail: (id: string) => [...checkoutKeys.all, id] as const,
 }
 
+export const shippingMethodKeys = {
```
Comment (kibo-kevinwatts) [inline lib/react-query/queryKeys.ts:8]: as stated above, never cache shipping methods so we don't need these

**Review by kibo-kevinwatts — COMMENTED** [review]
Niceee!

File: .env [hunk 4]
```diff
@@ -0,0 +1,17 @@
+KIBO_AUTH_HOST=home.mozu.com
+KIBO_CLIENT_ID=i7d6294.VueStorefront.1.0.0.Release
+KIBO_SHARED_SECRET=132213b896c3479592954ea6a92b6825
+KIBO_API_HOST=t26507-s41315.sandbox.mozu.com
+
```
[the remark this answers is not in the document]
Comment (kibo-sushant) [reply .env]: Resolved

File: components/checkout/ShippingStep/ShippingStep.tsx [hunk 5]
```diff
@@ -1,29 +1,132 @@
...
 }
 
 const ShippingStep = (props: ShippingProps) => {
-  const { stepperStatus, onCompleteCallback } = props
+  const { checkout } = props
+
+  const contact = checkout?.fulfillmentInfo?.fulfillmentContact as Contact
+
+  const orderItems = checkout?.items
+  const shipItems = orderItems?.filter((item) => item?.fulfillmentMethod === 'Ship')
```
[the remark this answers is not in the document]
Comment (kibo-sushant) [reply components/checkout/ShippingStep/ShippingStep.tsx]: Resolved

File: hooks/queries/useShippingMethods/useShippingMethods.tsx (hunk 2 above)
Comment (kibo-sushant) [reply hooks/queries/useShippingMethods/useShippingMethods.tsx]: Resolved using setting cacheTime: 0,

File: lib/react-query/queryKeys.ts (hunk 3 above)
Comment (kibo-sushant) [reply lib/react-query/queryKeys.ts:8]: Though I have set cacheTime: 0, we need key.

File: components/common/AddressForm/AddressForm.tsx (hunk 1 above)
Comment (kibo-kevinwatts) [reply components/common/AddressForm/AddressForm.tsx]: Remove the reference/import to the "CheckoutStepContext", otherwise it is good

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**